### PR TITLE
[eclipse/xtext#1952] converting test code to java

### PR DESCRIPTION
**Pull request by cdietrich**
[eclipse/xtext#1952] converting test code to java
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/formatting2/internal/CommentWithSpacesFormatterTest.java [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+				"\n" +
+				"\n" +
+				"a\n";
+		String output =
+				"idlist /*x*/\n" +
+				"a\n";
+		genericFormatterTester.assertFormatted((GenericFormatterTestRequest it) -> {
+			it.setToBeFormatted(input);
+			it.setFormatter(new TabsAndSpacesSupportingFormatter());
+			// Space before comment because of `oneSpace` and `addSpace` is
```
Comment (szarnekow) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/formatting2/internal/CommentWithSpacesFormatterTest.java:211]: Thank you for keeping the comments!

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/formatting2/internal/FormattableDocumentTest.java [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+			});
+			String expectation = "idlist a\n";
+			it.setExpectation(expectation);
+		});
+	}
+
+	@Test
+	public void autowrap() {
+		genericFormatterTester.assertFormatted((GenericFormatterTestRequest it) -> {
+			it.preferences((MapBasedPreferenceValues p) -> {
```
Comment (szarnekow) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/formatting2/internal/FormattableDocumentTest.java:65]: A few unnecessary types in lambda expressions, but I think it's fine.

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/xtext/XtextFormatterTest.java [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+				"	| '=>'\n" +
+				"	| '>' (=> ('>' '>') | '>')\n" +
+				"	| '<' (=> ('<' '<') | '<' | '=>')\n" +
+				"	| '<>'\n" +
+				"	| '?:';";
+		assertFormattedAs(Strings.toUnixLineSeparator(model), expected);
+	}
+
+	private void assertFormattedAs(CharSequence input, CharSequence expected) throws Exception {
+		Assert.assertEquals(Strings.toPlatformLineSeparator(expected).toString(), formattedText(input));
```
Comment (szarnekow) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/xtext/XtextFormatterTest.java:182]: It's not clear to me why we use `toUnixLineSeparator` for input and  `toPlatformLineSeparator` for expected.

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/xtext/XtextGrammarSerializationTest.java [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+				"\n" +
+				"NameRef returns xtext::RuleCall:\n" +
+				"	rule=[xtext::ParserRule];\n" +
+				"\n" +
+				"MyRule returns xtext::ParserRule:\n" +
+				"	name=ID;\n";
+		doTestSerialization(model, expectedModel);
+	}
+
+	public void _testXtestSerializationSelfTest() throws Exception {
```
Comment (szarnekow) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/xtext/XtextGrammarSerializationTest.java:192]: Is this used?

**Review by szarnekow — APPROVED** [review]

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/xtext/XtextFormatterTest.java (hunk 3 above)
Comment (cdietrich) [reply org.eclipse.xtext.tests/src/org/eclipse/xtext/xtext/XtextFormatterTest.java:182]: this is at it was in the original :(

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/xtext/XtextGrammarSerializationTest.java (hunk 4 above)
Comment (cdietrich) [reply org.eclipse.xtext.tests/src/org/eclipse/xtext/xtext/XtextGrammarSerializationTest.java:192]: no. i wanted to keep this for whatever it was good for.

**Pull request merged**